src/plots/cartesian/axes.js

```diff
@@ -1136,9 +1147,26 @@ function formatDate(ax, out, hover, extraPrecision) {
             }
         }
     }
-    if(suffix && (!ax._inCalcTicks || (suffix !== ax._prevSuffix))) {
-        tt += suffix;
-        ax._prevSuffix = suffix;
+    if(ax.tickmode === 'array') {
+        // we get extra precision in array mode, but it may be useless, strip it off
+        if(tt === '00:00:00') {
+            tt = suffix;
+            suffix = '';
+        }
+        else {
+            tt = tt.replace(/:00$/, '');
+        }
+    }
+
+    if(suffix) {
+        if(hover) {
+            // hover puts it all on one line, so suffix works best up front
+            tt = suffix + (tt ? ' ' + tt : '');
```

Comment:
Ah... yep, I see where that's coming from. Looks like I need to add some more explicit tests of `Axes.tickText` with hover; the regular (non-hover) cases should all be covered by the `calcTicks` test cases.